pkg/apis/clusteroperator/types.go

```diff
@@ -402,6 +391,21 @@ type ClusterDeploymentStatus struct {
 	// Kubernetes Cluster API controllers on the cluster.
 	ClusterAPIInstalledTime *metav1.Time
 
+	// ClusterVersionRef references the resolved clusterversion the cluster should be running.
+	// +optional
+	ClusterVersionRef *corev1.ObjectReference
+
+	// Ready is true if the master of the cluster is ready to be used and can be accessed using
+	// the KubeconfigSecret
+	Ready bool
+
+	// DeprovisionedComputeMachinesets is true if the compute machinesets of this cluster
+	// have been deprovisioned.
+	DeprovisionedComputeMachinesets bool
+}
+
+// ClusterDeploymentStatus contains the status for a cluster
+type ClusterDeploymentStatus struct {
 	// Ready is true if the master of the cluster is ready to be used and can be accessed using
 	// the KubeconfigSecret
 	Ready bool
```

Comment:
I don't know that we will reflect the status from the cluster here, I would just leave it on the ClusterProviderStatus for now